### PR TITLE
Adding first PainTracker questionnaires...

### PR DESCRIPTION
**Pull request by mcjustin**
... QuestionnaireResponses for these will be submitted from PainTracker (https://gitlab.cirg.washington.edu/svn/dhair2/-/merge_requests/597 ) to COSRI.

1. PEG for Pain
2. PainTracker body diagram.

Sample QR's for these are [here](https://fhir.acc.dev.cosri.cirg.washington.edu/fhir/QuestionnaireResponse?questionnaire=Questionnaire%2FCIRG-PEG) and [here](https://fhir.acc.dev.cosri.cirg.washington.edu/fhir/QuestionnaireResponse?questionnaire=Questionnaire%2FCIRG-PainTracker-Location-Body-Diagram).

### CODE REVIEW

File: CIRG-PEG.json [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        }
+      ]
+    },
+    {
+      "linkId": "91147-9",
+      "code": [
+        {
+          "system": "http://loinc.org",
+          "code": "91147-9",
+          "display": "Mean score [PEG]"
```
Comment (mcjustin) [inline CIRG-PEG.json:294]: TODO for me: add another item for "Summed score" (not standard PEG), as that's what we implemented in PT per request by the PT clinical team.

**Comment by mcjustin** [comment]
@achen2401  I now have both the PEG and pain location body diagram Questionnaires here, I think you were waiting for these...

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 2]
```diff
@@ -0,0 +1,1617 @@
...
+        {
+          "valueCoding": {
+            "code": "8447",
+            "display": "back_upper_right_posterior_thorax"
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8448",
+            "display": "front_right_anterior_neck"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:775]: @mcjustin I think this is `front_anterior_neck`.  I see no match for `front_right_anterior_neck` region on the diagram

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 3]
```diff
@@ -0,0 +1,1617 @@
...
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8449",
+            "display": "front_right_anterior_chest"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:781]: I think this is `front_anterior_chest`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 4]
```diff
@@ -0,0 +1,1617 @@
...
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8450",
+            "display": "front_right_anterior_upper_abdomen"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:787]: I think this is `front_anterior_upper_abdomen`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 5]
```diff
@@ -0,0 +1,1617 @@
...
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8451",
+            "display": "front_right_anterior_pelvic"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:793]: I think this is `front_anterior_pelvic`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 6]
```diff
@@ -0,0 +1,1617 @@
...
+        {
+          "valueCoding": {
+            "code": "8467",
+            "display": "back_lower_left_posterior_thorax"
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8468",
+            "display": "back_lower_right_posterior_thorax"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:1611]: I think this is `back_right_posterior_thorax`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 7]
```diff
@@ -0,0 +1,1617 @@
...
+        {
+          "valueCoding": {
+            "code": "8466",
+            "display": "back_right_posterior_head"
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8467",
+            "display": "back_lower_left_posterior_thorax"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:1605]: I think this is `back_left_posterior_thorax`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 8]
```diff
@@ -0,0 +1,1617 @@
...
+        {
+          "valueCoding": {
+            "code": "8465",
+            "display": "front_right_anterior_pelvic"
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8466",
+            "display": "back_right_posterior_head"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:1599]: I think this is `back_posterior_head`

File: CIRG-PainTracker-Location-Body-Diagram.json [hunk 9]
```diff
@@ -0,0 +1,1617 @@
...
+        {
+          "valueCoding": {
+            "code": "8464",
+            "display": "front_right_anterior_upper_abdomen"
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "8465",
+            "display": "front_right_anterior_pelvic"
```
Comment (achen2401) [inline CIRG-PainTracker-Location-Body-Diagram.json:1593]: I think this is `front_anterior_pelvic`

**Comment by achen2401** [comment]
@mcjustin I thought I'd check..coz I am not seeing some of the regions you listed in the body diagram questionnaire.  Are these the files I should be referencing:
https://gitlab.cirg.washington.edu/svn/dhair2/-/blob/master/app/webroot/img/body_diagram_paintracker_assessment_labels.svg (horizonal view of the body diagram)
https://gitlab.cirg.washington.edu/svn/dhair2/-/blob/master/app/webroot/img/body_diagram_paintracker.svg (vertical view of the body diagram)

File: CIRG-PainTracker-Location-Body-Diagram.json (hunk 2 above)
Comment (mcjustin) [reply CIRG-PainTracker-Location-Body-Diagram.json:775]: @achen2401  I'm pretty sure these left | right subdivisions that you've flagged here were added to dhair2 some years ago, but probably weren't in the original. 

I see them in dhair2 here:
```
grep -R front_right_anterior_chest app/ | grep -v sql | grep -v telepain
app/View/Surveys/body_diagram_paintracker.ctp:  <path class="transparent" d="M 168.97266 107.51562 C 161.15193 107.62465 153.79556 107.178 147.82031 106.45703 C 142.09511 115.76783 131.73373 129.63416 118.30273 134.91016 C 117.88513 135.07456 117.45332 135.22559 117.01172 135.36719 C 117.01172 135.36719 117.71562 150.97823 117.91602 150.36523 C 117.91602 150.36523 123.96009 174.03667 126.71289 189.38867 C 128.40129 198.80387 130.35759 206.19872 128.40039 216.76172 C 128.32719 217.15412 128.25981 217.52853 128.19141 217.89453 C 134.94381 218.14533 144.30008 217.94131 147.08008 215.16211 L 147.08008 215.16016 C 151.52162 210.71861 153.0675 174.16447 168.97266 172.52344 L 168.97266 107.51562 z" id="front_right_anterior_chest"/>
app/webroot/img/body_diagram_paintracker_report.svg:     id="front_right_anterior_chest"
```

Comment (achen2401) [reply CIRG-PainTracker-Location-Body-Diagram.json:775]: thanks, @mcjustin I will update the svg on my end to reflect the latest subdivisions.

**Comment by achen2401** [comment]
@mcjustin Not sure if you are still working on these, but I did try out the questionnaires you have here.
The following are the example mock questionnaire responses I tried: 
**PEG**:
```
{
        "resourceType": "QuestionnaireResponse",
        "id": "1618828",
        "meta": {
          "versionId": "1",
          "lastUpdated": "2023-05-17T12:02:44.186-04:00"
        },
        "questionnaire": "Questionnaire/CIRG-PEG",
        "status": "completed",
        "subject": {
          "reference": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162"
        },
        "authored": "2023-05-16T08:35:15-07:00",
        "author": {
          "reference": "Practitioner/52919099-6a7a-442c-b0d5-2b02c0dd4b74"
        },
        "item": [
          {
            "linkId": "108518",
            "text": "What number best describes your pain on average in the past week?",
            "answer": [
              {
                "valueCoding": {
                  "code": "75893-8-9",
                  "display": "9"
                }
              }
            ]
          },
          {
            "linkId": "91145-3",
            "text": "What number best describes how, during the past week, pain has interfered with your enjoyment of life?",
            "answer": [
              {
                "valueCoding": {
                  "system": "http://loinc.org",
                  "code": "LA10137-0",
                  "display": "5"
                }
              }
            ]
          },
          {
            "linkId": "91146-1",
            "text": "What number best describes how, during the past week, pain has interfered with your general activity?",
            "answer": [
              {
                "valueCoding": {
                  "system": "http://loinc.org",
                  "code": "LA10139-6",
                  "display": "7"
                }
              }
            ]
          },
          {
            "linkId": "91147-9",
            "text": "Mean score",
            "answer": [
              {
                "valueDecimal": 7
              }
            ]
          },
          {
            "linkId": "CIRG-PEG-SUM",
            "text": "Sum score",
            "answer": [
              {
                "valueDecimal": 21
              }
            ]
          }
        ]
      }
```
**body diagram**:
```
{
        "resourceType": "QuestionnaireResponse",
        "id": "1618809",
        "meta": {
          "versionId": "1",
          "lastUpdated": "2023-05-17T11:59:24.797-04:00"
        },
        "questionnaire": "Questionnaire/CIRG-PainTracker-Location-Body-Diagram",
        "status": "completed",
        "subject": {
          "reference": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162"
        },
        "authored": "2023-05-17T08:59:01-07:00",
        "author": {
          "reference": "Practitioner/52919099-6a7a-442c-b0d5-2b02c0dd4b74"
        },
        "item": [
          {
            "linkId": "CIRG-PainTracker-1252",
            "text": "Below is a list of locations of pain. Please indicate one or more areas where you have felt pain over the past week.",
            "answer": [
              {
                "valueCoding": {
                  "code": "6706",
                  "display": "front_right_top_of_foot"
                }
              },
              {
                "valueCoding": {
                  "code": "6712",
                  "display": "front_left_patellar_tendon"
                }
              }
            ]
          },
          {
            "linkId": "CIRG-PainTracker-1253",
            "text": "Please state precisely where your severe pain is",
            "answer": [
              {
                "valueCoding": {
                  "code": "6895",
                  "display": "back_right_lateral_foot"
                }
              }
            ]
          }
        ]
      }
```
and also the screenshot of what gets rendered in my instance:
PetBodyDiagram

**Comment by mcjustin** [comment]
@achen2401  I have these working in dhair2 now, so I'll merge this.

**Pull request merged**